include/quadrilateral_gimp_element.tcc

```diff
@@ -0,0 +1,155 @@
+// Return natural nodal coordinates
+template <>
+inline Eigen::MatrixXd
+    mpm::QuadrilateralGIMPElement<2, 16>::natural_nodal_coordinates() const {
+  //! Natural coordinates of nodes
+  Eigen::Matrix<double, 16, 2> local_nodes;
```

Comment:
Eigen has `.finished()` which allows you to define a const, could you please look at that.

Reply:
The idea is you can save this as a const